### DISCUSSION TITLE
Multitap for PS2 Support

### DISCUSSION THREAD

**Discussion by gitzapp** [darthcloud/BlueRetro#1074] [Ideas]
Does this mod support ps2 multitap? can i connect 4 BT controllers via BlueRetro? If this is not currently supported, is it possible to implement?

EDIT: For clarity... I am asking, can you connect multiple BT Controllers to a single BlueRetro Device, and have the BlueRetro Device act like what a PlayStation Multitap would do. I am not suggesting plugging 4 BlueRetro Receivers into a multi tap.

**Comment by patrikmckane** [comment]
Why dont you just try? you need 4x blueretro and 4 diferent controller paired to each blueretro. Thats all. Post results.

**Reply by gitzapp** [reply]
I think my initial comment might have been misunderstood. Let me try again.

Rather than only connecting 1 Bluetooth controller to the receiver, Will this receiver support connecting multiple controllers via bluetooth and can it support acting as a Multitap for the Bluetooth controllers.

**Comment by souzafl** [comment]
Friends, I have the same question. I know that it is possible to synchronize 4 joysticks, but only 2 work, because all the diagrams I found only show the connection of the 2 PS2 inputs. Would there be any way to use the other GPIOs, to connect to each Multitap input, and for it to use the 4 joysticks for multiplayer games? Just to tell you, my PS2 is model 50001, with an esp32 module placed internally.

**Comment by FatBaldDad** [comment]
The simple answer is no—well… not yet. I expect someone really smart (ahem… Darthcloud) will probably figure it out sooner or later.

If you think about what the multitap does, it actually has specific processes when in use. It interprets data from four controllers and sends it through a single port. Although I’m not exactly sure how it works, each controller essentially has an ID or address attached to its data, which is sent through the Player 1/mem card 1 ports. The console receives this data and parses it to the corresponding location for each controller, and the same applies for data sent from the console to the controllers.

You may already know this, but Blue Retro is a controller adapter (1 controller, 1 port, or the core, which supports 2 controllers and 2 ports). To support 4 controllers on one port, it would need to handle much more than its current capability (not downplaying Darthcloud's work—Blue Retro is a fantastic product, and I own two BR cores).

I believe there are some talented developers working on making a Blue Retro-compatible PS2 multitap a reality. That would be sweet.

**Comment by darthcloud** [comment]
Multitap is only supported for PS1.

PS2 multitap support require a connection to the memory card SPI bus which goes at 24MHz. The ESP32 slave SPI HW is limited to a maximum of 10MHz.

So it is not possible.

**Reply by FatBaldDad** [reply]
Aww
